[MIG] Migrate auditlog module from 8.0 to 9.0

holdenrehg opened the pull request:
The majority of the module was already following new api conventions and not using any deprecated methods, so there was only minor clean up to move to over to 9.0
- Update documentation to point to the new auditlog menu locations. These were changed because the 8.0 version was referencing menus that do not exist in 9.0
- Change version from 8.0.1.0.0 to 9.0.1.0.0
- Make the module installable again
- Remove an unused parameter from pre-migration.py for versioning
- Update classes to follow OCA new api conventions, camel case instead of snake case
- Fix typos and remove commented out blocks of code that were irrelevant

sebalix committed:
[IMP] Module 'auditlog' - Add 'name' attributes on some filters and other minor fixes

sebalix committed:
[IMP] Module 'auditlog' - Removed the 'timestamp' field from the 'auditlog.log' model (standard 'create_date' field is used instead)

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (based on the OCA template)

hbrunn committed:
[ADD] init hook to migrate from audittrail

hbrunn committed:
[IMP] ignore rules for nonexisting objects

hbrunn committed:
[FIX] also migrate id sequences

sebalix committed:
[IMP] Module 'auditlog' - Contributors updated in README.rst + AUTHORS.txt file removed

hbrunn committed:
[ADD] change xmlids of legacy rules to point to auditlog.rule

hbrunn committed:
[IMP] use new import of _, SUPERUSER_ID

hbrunn committed:
[ADD] allow overriding modules to pass additional log values

hbrunn committed:
[IMP] cache model and field ids

hbrunn committed:
[FIX] don't reset caches in register_hook

hbrunn committed:
[ADD] pass old_values to create_log when deleting a record

hbrunn committed:
[FIX] cope with no additional log values

hbrunn committed:
[ADD] tests

sebalix committed:
[FIX] Module 'auditlog' - local variable 'ir_model' is assigned to but never used

hbrunn committed:
[IMP] use a model for test that is not influenced by the mail module

sebalix committed:
[FIX] Module 'auditlog' - Remove 'active' field from __openerp__.py

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (inform that the audittrail data - if any - are migrated during the installation)

sebalix committed:
[IMP] Module 'auditlog' - Model 'auditlog.log.line', 'field_name' and 'field_description' fields changed into related fields

sebalix committed:
[IMP] Module 'auditlog' - Added a second test using cached data generated during the first test

sebalix committed:
[FIX] Module 'auditlog' - Data dictionaries of fields and models are put in the cache instead of recordsets (a recordset contains a reference on the database cursor which is closed after each RPC query)

sebalix committed:
[IMP] Module 'auditlog' - Add a test regarding a resource deleted through a (2, ID) tuple on a write operation

sebalix committed:
[FIX] Module 'auditlog' - When updating a resource, filter IDs on 2many fields to prevent a 'name_get()' call on potential deleted resources

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

sebalix committed:
[IMP] Module 'auditlog' - Performing logs on 'read' operations + Some bugfixes to log inherited fields and dummy fields such as 'in_group_X' in 'res.users' model +  Unit tests updated

sebalix committed:
[FIX] Module 'auditlog' - Replace 'try/except statement by 'isinstance()' + Add missing field in unit test + Remove a list comprehension

sebalix committed:
[FIX] Module 'auditlog' - Support 'read' calls with one ID or a list of IDs

sebalix committed:
[FIX] Module 'auditlog' - Unit tests, do not create users with the 'notify_email' field

eqms committed:
[ADD] German translation for auditlog

sebalix committed:
[FIX] Module 'auditlog' - No log for internal processing (e.g. 'read' calls produced by auditlog, either to scan records data or to fetch informations from 'ir.model'/'ir.model.fields' data models if rules are defined on them)

sebalix committed:
[FIX] Module 'auditlog' - Fix lint check

sebalix committed:
[IMP] Module 'auditlog' - Unsubscribe rules before removing them + Unit tests updated

sebalix committed:
[FIX] Module 'auditlog' - Log fields coming from polymorphic inheritances ('_inherits')

sebalix committed:
[IMP] Module 'auditlog' - Remove the unit test causing trouble with Travis CI

Commit:
[ADD] auditlog: Added Spanish translation.

yvaucher committed:
Add bug tracker link on README.rst

hbrunn committed:
[FIX] ie related fields don't have an

ir.model.fields entry

hbrunn committed:
[ADD] comment

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebalix committed:
[IMP] Module 'auditlog' - Log HTTP user sessions and requests

sebalix committed:
[IMP] Module 'auditlog' - Bump the version to 1.1.0

sebalix committed:
[FIX] Module 'auditlog' - Cleanup XML views (types removed)

sebalix committed:
[FIX] Module 'auditlog' - A log can be created with no current HTTP request (unit tests, ir.cron...)

sebalix committed:
[FIX] Module 'auditlog' - 'display_name' fields added for the user session and HTTP requests models

sebalix committed:
[FIX] Module 'auditlog' - Fix pylint check

hbrunn committed:
[FIX] turn off prefetching when reding values

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebalix committed:
[IMP] auditlog - Ability to choose the log type on the rule: Full log (complete but slow) and Fast log (data input only, faster)

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebalix committed:
[FIX] auditlog - Field 'type' renamed to 'log_type'

sebalix committed:
[IMP] auditlog - Auto-vacuum logs, HTTP requests and HTTP user sessions

sebalix committed:
[IMP] auditlog - Bump version number

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebalix committed:
[IMP] auditlog - Icon added + README updated (screenshots)

oca-transbot committed:
OCA Transbot updated translations from Transifex

holdenrehg committed:
[MIG] Migrate auditlog module from 8.0 to 9.0

- Update documentation to point to the new auditlog menu locations. These were changed because the 8.0 version was referencing menus that do not exist in 9.0
- Change version from 8.0.1.0.0 to 9.0.1.0.0
- Make the module installable again
- Remove an unused parameter from pre-migration.py for versioning
- Update classes to follow OCA new api conventions, camel case instead of snake case
- Fix typos and remove commented out blocks of code that were irrelevant

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

holdenrehg commented:
Here is a reference to my CLA signature https://github.com/odoo/odoo/pull/12305

moylop260 commented:
@holdenrehg 
Thanks for you contribution.

> Here is a reference to my CLA signature odoo/odoo#12305

This one is CLA of odoo. Could you sign your CLA of oca https://odoo-community.org/page/cla too?

moylop260 commented:
Could you check the [license](https://travis-ci.org/OCA/server-tools/jobs/137134463#L438)?
- `auditlog/__openerp__.py:22: [C8102(manifest-required-key), ] Missing required key "license" in manifest file`

More info [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules)

holdenrehg committed:
[UPD] Updated the __openerp__.xml to include the license and images fields

These were updated to follow OCA conventions.

- License set to AGPL-3
- Images set to empty array

holdenrehg commented:
@moylop260 Checked and updated the license field. [See here](https://github.com/OCA/server-tools/pull/454/commits/233bb87ba5ef636cfd6ef95ab5b955d4f1293a4a) for the commit.

moylop260 commented:
Thank you
After CLA detail... LGTM

hbrunn commented:
you should follow the procedure outlined in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#howto to get 8.0's commits. This allows us to simply cherry pick changes afterwards and keeps history intact.

holdenrehg commented:
Sounds good @hbrunn I'll pull in that 8.0 history and update the PR.

holdenrehg commented:
@hbrunn It appears that all the 8.0 commit history is already there. There was an updated translations file that I pulled from the 12th after I migrated the module.

hbrunn commented:
you seem to run the commands mentioned above on your current branch, this won't give you the desired results. You need to start out from an upstream 9.0 branch, and then do do the stuff so that the commit history includes all of those: https://github.com/OCA/server-tools/commits/8.0/auditlog

I also see that there are some commits that make the module better, but are not reflected in this PR, like https://github.com/OCA/server-tools/commit/3f9152ccbefabe3b4802e2944ab5040eb0fad106

holdenrehg committed:
[UPD] auditlog module - Merge commit from 8.0

holdenrehg commented:
@hbrunn So is the proper way to perform the actions on an upstream branch and then merge into this PR? It will add 70+ commits to this PR. Want to make sure I'm handling this the correct way according to OCA conventions.

hbrunn commented:
yes, that's exactly what we want

Just do all of this in a fresh branch and push overwrite this one. Or just close this PR and create a new one with your new branch.

holdenrehg commented:
Ah thanks @hbrunn

sebalix commented on auditlog/README.rst:
```diff
@@ -1,29 +1,42 @@
+
+Get the details:
+
+.. image:: /auditlog/static/description/log.png
```
I set all these image URLs in my last PR, but it seems they do not work properly on apps.odoo.com (https://apps.odoo.com/apps/modules/8.0/auditlog/). We should change them to relative ones like `static/description/IMAGE.png` in order to have them displayed.

holdenrehg replied:
Relative paths do not work when testing. I'll leave these as absolute for now.

hbrunn commented on auditlog/__openerp__.py:
```diff
@@ -1,38 +1,27 @@
     'author': "ABF OSIELL,Odoo Community Association (OCA)",
+    'license': "AGPL-3",
     'website': "http://www.osiell.com",
+    'license': 'AGPL-3',
```
two is a crowd already: https://travis-ci.org/OCA/server-tools/jobs/137559872#L433

hbrunn commented:
runbot error unrelated

holdenrehg committed:
[UPD] auditlog module, Remove duplicate license key from openerp config

holdenrehg commented:
@hbrunn stripped out that duplicated license key, also thanks for mentioning unrelated runbot errors. Saw the same, related to other modules.

sebalix commented:
Tested on Runbot, :+1:

sebalix commented on auditlog/migrations/8.0.1.0/pre-migration.py:
```diff
@@ -1,26 +1,9 @@
 
 
-def migrate(cr, version):
+def migrate(cr):
```
I did not see this. We should not change the signature of the method here, even if the version parameter is not used. It seems that the migrate method is always called with this parameter: https://github.com/odoo/odoo/blob/9.0/openerp/modules/migration.py#L152

pedrobaeza replied:
In fact, this file should be totally removed, as the migration doesn't apply to the version 9.

sebalix replied:
It's right.

holdenrehg committed:
[DEL] auditlog: Remove migration from audittrail functionality, this is not required for version 9

holdenrehg committed:
[DEL] auditlog: Remove pre_init_hook reference from openerp, no pre_init hook exists any more

moylop260 commented on auditlog/models/http_request.py:
```diff
@@ -0,0 +1,60 @@
+    _order = "create_date DESC"
+    _rec_name = 'display_name'
+
+    display_name = fields.Char(u"Name", compute="_display_name")
```
`s/_display_name/_compute_display_name/g`

sebalix replied:
Is this a blocking point to migrate on 9.0? At best rename `_display_name` to `_compute_display_name`.

hbrunn commented on auditlog/models/http_request.py:
```diff
@@ -0,0 +1,60 @@
+    _order = "create_date DESC"
+    _rec_name = 'display_name'
+
+    display_name = fields.Char(u"Name", compute="_display_name")
```
any reason not to overwrite `name_get` instead? This will be the display name by default, with the added benefit that `name_get` also returns something sensible. Then you also don't need to set `_rec_name`. cf https://github.com/OCA/OCB/blob/9.0/openerp/models.py#L1664

holdenrehg replied:
What's better in this scenario? `name_get` seems to be less elegant if you still keep the computed `display_name` field. If you get rid of `display_name` and compute directly in `name_get` that has worse performance since it will be computed on every view load that has this model defined.

``` python
class AuditlogHTTPRequest(models.Model):

    ...

    display_name = fields.Char(u"Name", compute="_compute_display_name")

    @api.multi
    @api.depends('create_date', 'name')
    def _compute_display_name(self):
        for httprequest in self:
            create_date = fields.Datetime.from_string(httprequest.create_date)
            tz_create_date = fields.Datetime.context_timestamp(
                httprequest, create_date)
            httprequest.display_name = u"%s (%s)" % (
                httprequest.name or '?',
                fields.Datetime.to_string(tz_create_date))

    @api.multi
    def name_get(self):
        res = []
        for request in self:
            res += [(request.id, request.display_name)]
        return res
```

pedrobaeza commented:
Please rebase the branch to make it mergeable

pedrobaeza commented:
Superseeded by #603
